lib/Auth/Process/checkEntityID.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * This Auth Proc Filter allows the selective deactivation of privacyIDEA for a list of regular expressions
+ * which match SAML service provider entityIDs.
+ * The filter checks the entityid in the SAML request against a list of regular expressions and sets the state variable
+ * $state[enabledPath][enabledKey][0] to false on match, which can be used to disable privacyIDEA.
+ * For any value in entityids, the config parameter excludeattributes may be used to enable privacyIDEA for a subset
+ * of users which have these attribute values (e.g. memberOf).
+ * @author Henning Hollermann <henning.hollermann@netknights.it>
+ */
+
+class sspmod_privacyIDEA_Auth_Process_checkEntityID extends SimpleSAML_Auth_ProcessingFilter {
+
+    private $entityids = array();
+    private $excludeattributes = array();
+    private $setPath = '';
+    private $setKey = '';
+
+    private function str_matches_reg_arr($str, $reg_arr) {
+        /*
+         * This function checks a given string against an array with regular expressions
+         * It will return true if a match is found.
```

Comment:
This is not quite right. Actually it returns an array, so which on is it or should it be?

Reply:
That's true, I changed the function but not the docstring. Thank's.